src/main/java/org/folio/circulation/domain/LoanService.java

```diff
@@ -30,24 +30,25 @@ public LoanService(Clients clients) {
     if(requests.isEmpty()) {
       return completedFuture(succeeded(records));
     }
-    /*
-      This gets the top request, since UpdateRequestQueue.java#L106 updates the request queue prior to loan creation.
-      If that sequence changes, the following will need to be updated to requests.stream().skip(1).findFirst().orElse(null)
-      and the condition above could do a > 1 comparison. (CIRC-277)
-    */
-    Request nextRequestInQueue = requests.stream().findFirst().orElse(null);
-    if (nextRequestInQueue == null || nextRequestInQueue.getRequestType() != RequestType.RECALL) {
+    
+    //check to see if there are any recalls in the Queue
+    Boolean isRecallInQueue = requests.stream().anyMatch(request -> request.getRequestType() == RequestType.RECALL);
+
+    //if no, exit 
+    if (!isRecallInQueue) {
```

Comment:
Could this be encapsulated into the RequestQueue class?

I'd like to try to minimise the access to the collection of requests in the queue themselves where possible, as this makes it easier to change how the request queue works in the future and I think it describes intent more succinctly.

```suggestion
    if (!requestQueue.containsRequestOfType(RECALL)) {
```